docs/docs/Architecture.KeyConstructs.Overview.document.md

```diff
@@ -73,6 +63,38 @@ The two key characteristics of a Flight Recorder are:
 1. that it is uniquely identifiable
 1. that it contains logs
 
+## What is a Filter/Router/Adapter{.unlisted .unnumbered}
+
+Imagine an OpenTelemetry pipeline as a pipe of water. Cutting the pipe and
+adding a filter or diverter would allow you to control the flow. A fully
+blocking filter could be considered a valve. The use of filters and routers
+allows the dynamic shaping and routing of telemetry - much like valves and
+filters in a water system.
+
+* A filter is a Dynamic Telemetry construct used to filter and route logging
+  that is already inside an OpenTelemetry pipeline.
+* A Router is a dynamic telemetry construct that allows the data pipeline to be
+  cut and forked. For example, it can connect two different processors, perhaps
+  one that streams into an OpenTelemetry backend while the other goes into a
+  Flight Recorder.
+* An Adapter allows the adaptation of existing non-OpenTelemetry and often
+  platform-specific telemetry into the OpenTelemetry pipeline. For example,
+  syslog, LTTng, or ETW on Windows.
+
+## What is a Processor{.unlisted .unnumbered}
+
+A Processor is the dynamic component of Dynamic Telemetry where various
```

Comment:
What does "dynamic component" mean?

Reply:
something that can change based on situation and not just config - for example "I see the queue depths are >1000000, lets use the CPU sampling 'Action'"

Reply:
p.s. I may want a better than than 'Processor' due to name collision in OT?  ("Dynamic Processor"?)  - by definition it'd be one that contains one of the three "Processor Types"

hey; if you want, feel free to point 3P friends you have in OT to this PR?

Reply:
Mmm I think we might actually need a term for the elements that connect pipelines. "Connector" is a term that the current OTel Collector uses, but it's not the term it uses for the substantially-complex logic that deals with queueing, batching, and concurrency--what the OTel Collector refers to as exporter helpers we may need alternate terminology for.

@chgray We have some parallel thinking in the OTel-Arrow project, here's an active draft which discusses the same type of dynamic connector topic: https://github.com/lquerel/otel-arrow/blob/dataflow-doc/rust/beaubourg/docs/dataflow-design.md#admission-control